packages/react-core/src/components/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import styles from '@patternfly/patternfly-next/components/Dropdown/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import DropdownMenu from './DropdownMenu';
+import DropdownToggle from './DropdownToggle';
+
+export const DropdownPosition = {
+  right: 'right',
+  left: 'left'
+};
+
+export const DropdownDirection = {
+  up: 'up',
+  down: 'down'
+};
+
+const propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
+  isCollapsed: PropTypes.bool,
+  position: PropTypes.oneOf(Object.keys(DropdownPosition)),
+  direction: PropTypes.oneOf(Object.keys(DropdownDirection)),
+  isKebab: PropTypes.bool,
+  title: PropTypes.node,
+  hasArrow: PropTypes.bool,
+  onSelect: PropTypes.func,
+  airaLabelledBy: PropTypes.string,
+  ariaLabel: PropTypes.string,
+  onToggle: PropTypes.func
+};
+
+const defaultProps = {
+  title: null,
+  children: null,
+  className: '',
+  isCollapsed: true,
+  position: DropdownPosition.left,
+  direction: DropdownDirection.down,
+  isKebab: false,
+  hasArrow: true,
+  ariaLabel: undefined,
+  airaLabelledBy: null,
+  onSelect: Function,
+  onToggle: Function
+};
+
+class Dropdown extends Component {
+  constructor(props) {
+    super(props);
+    this.onDropdownClick = this.onDropdownClick.bind(this);
```

Comment:
I come from Java world and honestly I don't really like this way how methods are written (not that binding is better). But sure thing I'll use this syntax. Btw what do you think about [autobind-decorator](https://www.npmjs.com/package/autobind-decorator)?

Reply:
Not a huge fan of decorators honestly since they are stage2. It is usually safer to stay in the Stage 3+ world. As for `autobind` itself, I think it is better to use the actual platform instead. It will likely be optimized in the future and follows how the React comunity as a whole is declaring these. 

Comparing to bind, I put together a perf test which actually shows class properties coming out ahead as the fastest. I put it together when we were trying to deprecate `bindMethods`.
https://jsperf.com/class-property-vs-bind